nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -322,128 +326,52 @@ def add_electrodes(
     for prop in property_names:
-        prop_skip = False
         if prop not in exclude_names:
-            data = []
-            prop_chan_count = 0
-            # build data:
-            for chan_id in checked_recording.get_channel_ids():
-                if prop in property_names:
-                    prop_chan_count += 1
-                    chan_data = checked_recording.get_channel_property(channel_id=chan_id, key=prop)
-                    # find the type and store (only when the first channel with given property is found):
-                    if prop_chan_count == 1:
-                        proptype = [
-                            proptype for proptype in channel_property_defaults if isinstance(chan_data, proptype)
-                        ]
-                        if len(proptype) > 0:
-                            found_property_types[prop] = proptype[0]
-                            # cast as float if any number:
-                            if found_property_types[prop] == Real:
-                                chan_data = float(chan_data)
-                            # update data if wrong datatype items filled prior:
-                            if len(data) > 0 and not isinstance(data[-1], found_property_types[prop]):
-                                data = [channel_property_defaults[found_property_types[prop]]] * len(data)
-                        else:
-                            prop_skip = True  # skip storing that property if not of default type
-                            break
-                    data.append(chan_data)
-                else:
-                    data.append(channel_property_defaults[found_property_types[prop]])
-            # store data after build:
-            if not prop_skip:
-                index = found_property_types[prop] == ArrayType
-                prop_name_new = "location" if prop == "brain_area" else prop
-                found_property_types[prop_name_new] = found_property_types.pop(prop)
-                elec_columns[prop_name_new].update(description=prop_name_new, data=data, index=index)
+            data = checked_recording.get_property(prop)
+            # store data after build and remap some properties to relevant nwb columns:
+            if prop == "location":
+                location_map = ["rel_x", "rel_y", "rel_z"]
+                for prop_name_new, loc in zip(location_map, range(data.shape[1])):
+                    elec_columns[prop_name_new].update(description=prop_name_new, data=data[:, loc], index=False)
+            else:
+                if prop == "brain_area":
+                    prop = "location"
+                elif prop == "group":
+                    if "group_name" not in property_names:
+                        prop = "group_name"
+                    else:
+                        continue
+                index = isinstance(data[0], (list, np.ndarray))
+                elec_columns[prop].update(description=prop, data=data, index=index)
```

Comment:
This entire block looks like it might be OK - @h-mayorquin can you separate this block into a separate sub-PR?